packages/react-dialog/src/stories/AlertDialog.stories.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { Button } from '@fluentui/react-button';
+import { Dialog, DialogProps, DialogHeader, DialogBody, DialogFooter } from '../index';
+
+export const AlertDialog = (props: Partial<DialogProps>) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  return (
+    <>
+      <Button
+        onClick={() => {
+          setIsOpen(true);
+        }}
+        aria-haspopup="true"
```

Comment:
This can become a frequent bug where people forget to add aria-haspopup. Trigger component would be a good abstraction to use, and it could still be combined with the controlled state that you propose.

Reply:
Also, `aria-haspopup="dialog"` might be more accurate